tests/system/api/test_artifacts.py

```diff
@@ -58,3 +61,70 @@ def test_fail_overflowing_artifact(self):
         assert (
             "Failed committing changes to DB" in run["status"]["error"]
         ), "run should fail with a reason"
+
+    def test_model_artifact_tags(self):
+        project_name = "my-test-project"
+        model_name = "test_model"
+        tag_1 = "tag_1"
+        tag_2 = "tag_2"
+        models_path = f"v3io:///projects/{project_name}/artifacts/{model_name}/"
+
+        def _validate_artifact(expected_tags):
+
+            # get the model from api, and check that the tags are as expected
+            response = self._run_db.api_call(
+                "GET",
+                f"projects/{project_name}/artifacts?category=model&format=full&best-iteration=true",
+            )
+            assert response.status_code == 200, "failed to get artifacts"
+            for artifact in response.json().get("artifacts", []):
+                tag = artifact.get("metadata").get("tag")
+                assert tag in expected_tags, "given tag is not in expected tags"
+
+                # make sure artifact doesn't have the model_spec.yaml in spec.extra_data
+                extra_data = artifact.get("spec", {}).get("extra_data")
+                if extra_data:
+                    assert (
+                        "model_spec.yaml" not in extra_data
+                    ), "model spec should not be in extra data"
+
+            # get the model spec via sdk, and check that the tag is there
+            _, model_obj, _ = mlrun.artifacts.get_model(models_path)
+            model_dict = model_obj.to_dict()
+            assert (
+                "tag" in model_dict["metadata"]
+            ), "tag should not be in model metadata"
+
+        try:
+            pickle_filename = "my_model.pkl"
+            with open(pickle_filename, "wb") as file:
+                pickle.dump({"obj": "my-model"}, file)
+
+            # create a project and log the model
+            mlrun.set_environment(project=project_name)
+            project = mlrun.get_or_create_project(project_name, context="./")
+
+            project.set_model_monitoring_credentials(os.environ.get("V3IO_ACCESS_KEY"))
+            project.log_model(
+                model_name,
+                body="model body",
+                model_file="trained_model.pkl",
+                tag=tag_1,
+            )
+
+            _validate_artifact(["latest", tag_1])
+
+            # tag the model with another tag
+            _, model_obj, _ = mlrun.artifacts.get_model(models_path)
+            db = mlrun.get_run_db()
+            db.tag_artifacts(
+                artifacts=[model_obj],
+                project=project_name,
+                tag_name=tag_2,
+                replace=False,
+            )
+
+            _validate_artifact(["latest", tag_1, tag_2])
```

Comment:
not sure what this validates
I prefer if you would validate the exact tag instead of checking that a tag is one of many

Reply:
When we tag an artifact we have 1 artifact per tag, so I wanted to make sure they all exist.
I changed the validation function accordingly.